src/arch/src/lib.rs

```diff
@@ -319,10 +338,35 @@ fn create_rtc_node(fdt: &mut FdtWriter, rtc_addr: u64, size: u64) -> Result<()>
     Ok(())
 }
 
+fn create_virtio_node(fdt: &mut FdtWriter, addr: u64, size: u64, irq: u32) -> Result<()> {
+    let virtio_mmio = fdt.begin_node(&format!("virtio_mmio@{:x}", addr))?;
+    fdt.property_string("compatible", "virtio,mmio")?;
+    fdt.property_array_u64("reg", &[addr, size])?;
+    fdt.property_array_u32(
+        "interrupts",
+        &[GIC_FDT_IRQ_TYPE_SPI, irq, IRQ_TYPE_EDGE_RISING],
+    )?;
+    fdt.property_array_u32("interrupt-parent", &[PHANDLE_GIC])?;
+    fdt.end_node(virtio_mmio)?;
+    Ok(())
+}
+
 #[cfg(test)]
 mod tests {
     use crate::FdtBuilder;
 
+    #[test]
+    fn test_adding_virtio() {
+        let mut fdt = FdtBuilder::new();
+        let fdt = fdt.add_virtio_device(0x1000, 1000, 5);
+        assert_eq!(fdt.virtio_devices.len(), 1);
+    }
+    #[test]
+    fn test_virtio_device_len() {
```

Comment:
This test and the one above seem to be doing the same thing. Can you delete one of them?